### PR TITLE
Migrate to Custom Connect Modal

### PR DESCRIPTION
**Pull request by devstein**
### Description

This PR
- Creates a custom Picket `ConnectModal`
- Migrates off `Web3Modal`
- Creates a consistent interface for both EVM and Solana wallets
- Defaults `picket.login` to show all available chains
  
This is the first of many revisions to the Picket `ConnectModal`. Future work includes:

- [ ] Authenthicating/Authorizing users before the modal closes
- [x] Timeout warning if user hasn't connect their wallet in over 15 seconds
- [ ] Two-step flow with custom QR code
- [ ] `New Wallet` button
- [ ] Refactoring this out into a separate `@picketapi/picket-connect` library
- [ ] And more!

### CODE REVIEW

**Comment by devstein** [comment]
@noahfradin I think this version is good enough to merge. It matches `Web3Modal` functionality + doesn't close until the user has signed the nonce. 

I can show you how to test locally or I can merge and make a preview deployment of picketapi.com with the new modal

**Comment by devstein** [comment]
@noahfradin Process is 

1. Checkout this branch
2. Run
```shell
npm install
npm link
```
3. Create and setup new app locally (i.e NextJS project)
4. After setting up, run

```bash
npm link "@picketapi/picket-js"
```

Let me know if you run into issues

File: src/connect/ConnectModal.tsx [hunk 1]
```diff
@@ -328,7 +330,7 @@ const ConnectModal = ({
           )}
         </div>
         <div
-          className={tw`w-full mt-8 text-center font-light text-sm text-gray-500`}
+          className={tw`w-full mt-8 text-center font-regular text-sm text-gray-400`}
```
Comment (devstein) [inline src/connect/ConnectModal.tsx:333]: Should be `font-base`

File: src/connect/ConnectModal.tsx [hunk 2]
```diff
@@ -317,9 +319,9 @@ const ConnectModal = ({
                   clipRule="evenodd"
                 />
               </svg>
-              <div className={tw`flex flex-col items-center text-center space-y-4`}>
+              <div className={tw`flex flex-col items-center text-center space-y-2`}>
                 {selectedWallet?.icon}
-                <p className={tw`text-center`}>
+                <p className={tw`text-sm font-regular w-45 text-center text-gray-400 break-normal`}>
```
Comment (devstein) [inline src/connect/ConnectModal.tsx:324]: Same font-base

File: src/connect/ConnectModal.tsx [hunk 3]
```diff
@@ -209,21 +209,21 @@ const ConnectModal = ({
...
         <h1
-          className={tw`pt-2 text-2xl font-bold ${
+          className={tw`pt-2 text-xl font-semibold ${
             success ? "text-center" : "text-left"
           } `}
         >
           {success
             ? `Welcome ${displayAddress(walletAddress as string)}`
-            : "Sign-In with Your Wallet"}
+            : "Login with your wallet"}
```
Comment (devstein) [inline src/connect/ConnectModal.tsx:221]: Why the change in copy? Not opposed, but think we should be consistent with https://picketapi.com/

**Review by devstein — COMMENTED** [review]
@noahfradin

File: src/connect/ConnectModal.tsx (hunk 1 above)
Comment (noahfradin) [reply src/connect/ConnectModal.tsx:333]: Why?

File: src/connect/ConnectModal.tsx (hunk 3 above)
Comment (noahfradin) [reply src/connect/ConnectModal.tsx:221]: I'm open to Sign In or Login. Don't think I have a strong preference... Just no dashes.

Comment (noahfradin) [reply src/connect/ConnectModal.tsx:221]: Do you have a preference? Any reasoning behind it?

File: src/connect/ConnectModal.tsx (hunk 2 above)
Comment (noahfradin) [reply src/connect/ConnectModal.tsx:324]: Do you mean font-normal?

File: src/connect/ConnectModal.tsx (hunk 1 above)
Comment (noahfradin) [reply src/connect/ConnectModal.tsx:333]: Do you mean 'font-normal'?

File: src/connect/ConnectModal.tsx (hunk 3 above)
Comment (devstein) [reply src/connect/ConnectModal.tsx:221]: I opt-ed for consistency with https://login.xyz/

File: src/connect/ConnectModal.tsx (hunk 1 above)
Comment (devstein) [reply src/connect/ConnectModal.tsx:333]: yes

File: src/connect/ConnectModal.tsx (hunk 3 above)
Comment (devstein) [reply src/connect/ConnectModal.tsx]: Let's do log in. I followed the NY Times title case convention for capitalization
